client/my-sites/plugins/plugin-activate-toggle/index.jsx

```diff
@@ -1,87 +1,95 @@
+		PluginsActions.togglePluginActivation( site, plugin );
+		PluginsActions.removePluginsNotices( notices.completed.concat( notices.errors ) );
 
-		if ( this.props.plugin.active ) {
-			analytics.ga.recordEvent( 'Plugins', 'Clicked Toggle Deactivate Plugin', 'Plugin Name', this.props.plugin.slug );
-			analytics.tracks.recordEvent( 'calypso_plugin_deactivate_click', {
-				site: this.props.site.ID,
-				plugin: this.props.plugin.slug
+		if ( plugin.active ) {
+			this.props.recordGoogleEvent( 'Plugins', 'Clicked Toggle Deactivate Plugin', 'Plugin Name', plugin.slug );
+			this.props.recordTracksEvent( 'calypso_plugin_deactivate_click', {
+				site: site.ID,
+				plugin: plugin.slug
 			} );
 		} else {
-			analytics.ga.recordEvent( 'Plugins', 'Clicked Toggle Activate Plugin', 'Plugin Name', this.props.plugin.slug );
-			analytics.tracks.recordEvent( 'calypso_plugin_activate_click', {
-				site: this.props.site.ID,
-				plugin: this.props.plugin.slug
+			this.props.recordGoogleEvent( 'Plugins', 'Clicked Toggle Activate Plugin', 'Plugin Name', plugin.slug );
+			this.props.recordTracksEvent( 'calypso_plugin_activate_click', {
+				site: site.ID,
+				plugin: plugin.slug
 			} );
 		}
-	},
+	}
 
-	render: function() {
-		var inProgress;
+	render() {
+		const { site, plugin, isMock, disabled, translate } = this.props;
 
-		if ( ! this.props.site ) {
+		if ( ! site ) {
 			return null;
 		}
 
-		inProgress = PluginsLog.isInProgressAction( this.props.site.ID, this.props.plugin.slug, [
+		const inProgress = PluginsLog.isInProgressAction( site.ID, plugin.slug, [
 			'ACTIVATE_PLUGIN',
 			'DEACTIVATE_PLUGIN'
 		] );
 
-		if ( this.props.plugin && 'jetpack' === this.props.plugin.slug ) {
+		if ( plugin && 'jetpack' === plugin.slug ) {
 			return (
 				<PluginAction
 					className="plugin-activate-toggle"
-					htmlFor={ 'disconnect-jetpack-' + this.props.site.ID }
+					htmlFor={ 'disconnect-jetpack-' + site.ID }
 					>
 					<DisconnectJetpackButton
-						disabled={ this.props.disabled || ! this.props.plugin }
-						site={ this.props.site }
+						disabled={ disabled || ! plugin }
+						site={ site }
 						redirect="/plugins/jetpack"
-						isMock={ this.props.isMock }
+						isMock={ isMock }
 						/>
 				</PluginAction>
 			);
 		}
 		return (
 			<PluginAction
-				disabled={ this.props.disabled }
+				disabled={ disabled }
 				className="plugin-activate-toggle"
-				label={ this.translate( 'Active', { context: 'plugin status' } ) }
+				label={ translate( 'Active', { context: 'plugin status' } ) }
 				inProgress={ inProgress }
-				status={ this.props.plugin && this.props.plugin.active }
+				status={ plugin && plugin.active }
 				action={ this.toggleActivation }
-				htmlFor={ 'activate-' + this.props.plugin.slug + '-' + this.props.site.ID }
+				htmlFor={ 'activate-' + plugin.slug + '-' + site.ID }
 				/>
 		);
 	}
-} );
+}
+
+PluginActivateToggle.propTypes = {
+	site: PropTypes.object.isRequired,
```

Comment:
>  Not sure if this is a good idea,

It is 😄 but I can understand if there's too much to untangle at once. Usually, it's a bit easier if we update child components to handle their own data needs first. @tyxla feel free to make a judgement call on if you want to clean up the site passing here or in a follow up PR.